src/main/java/org/purescript/ide/inspections/PSUnresolvedReferenceInspection.kt

```diff
@@ -46,6 +49,27 @@ class PSUnresolvedReferenceInspection : LocalInspectionTool() {
                 }
             }
 
+            private fun visitTypeReference(reference: PsiReference) {
+                if (reference.canonicalText in PSLanguage.BUILTIN_TYPES) {
+                    return
+                }
+
+                // TODO Workaround to prevent false positives on class constraints
+                val isClassConstraint = reference.element.parent
+                    .siblings(forward = true, withSelf = false)
+                    .any { it.text == "=>" }
+                if (isClassConstraint) {
+                    return
+                }
+
+                // TODO Workaround to prevent false positives on qualified types
+                if (reference.element.textContains('.')) {
+                    return
+                }
```

Comment:
12 lines of workarounds... Hopefully we can remove this when we improve the parser.